sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -461,51 +628,310 @@ is `name`.
 
 _Throws:_
 
+* An `exception` with the `errc::invalid` error code if
+  `ext_oneapi_has_kernel(name)` returns `false`.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+std::string ext_oneapi_get_raw_kernel_name(const std::string &name)
+----
+!====
+
+_Constraints:_ This function is not available when `State` is
+`bundle_state::ext_oneapi_source`.
+
+_Returns:_ If the kernel bundle was created from a bundle of state
+`bundle_state::ext_oneapi_source` and `name` was registered via
+`registered_kernel_names`, returns the compiler-generated (e.g. mangled) name
+for this kernel function.
+If the kernel bundle was created from a bundle of state
+`bundle_state::ext_oneapi_source` and `name` is the same as a
+compiler-generated name for a kernel defined in that bundle, that same
+`name` is returned.
+
+_Throws:_
+
 * An `exception` with the `errc::invalid` error code if
   `ext_oneapi_has_kernel(name)` returns `false`.
 |====
 
+=== Obtaining a kernel when the language is `sycl`
```

Comment:
I am wondering what would be the behavior in case the user query were ambiguous, e.g.:
```c++
void foo(int);
void foo(float);
```
Now, querying for `"foo"` would be ambiguous, as both `void foo(int)` and `void foo(float)` apply. Do we need to document this case?

Reply:
We use normal C++ rules to disambiguate the two overloads.  The formal requirement is that the string passed to `registered_kernel_names` is an expression for a function pointer.  The way to do this in C++ when there are different overloads of the function is to use a cast.  This is illustrated in the examples section "Disambiguating overloaded kernel functions" below.